src/Adapter/ArrayAdapter.php

```diff
@@ -21,17 +28,33 @@ public function send($queue, $message)
         return $d['id'];
     }
 
-    public function deleteMessage($receipt, $queue)
+    public function receive($queue)
     {
+        if (!array_key_exists($queue, $this->data)) {
+            throw new OutOfBoundsException("Missing data");
+        }
+
+        $values = array_values($this->data[$queue]);
```

Comment:
This method could be written better than this because i am mixing array_values with array_shift...